### PR TITLE
Update change log in preparation for v3.0.0

### PR DESCRIPTION
**Pull request by bdash**
I rolled up the change log entries from the prerelease versions of v3.0.0 into the v3.0.0 entry to make it possible to see the changes in one place.

### CODE REVIEW

**Comment by bdash** [comment]
I should mention that I attempted to order entries roughly by user interest / applicability (i.e., arrays of primitives is at the top of the Enhancements section).

**Comment by bdash** [comment]
The Breaking Changes section is dominated by renames. Perhaps we should use a table for renames instead?

**Comment by austinzheng** [comment]
A table sounds like a good idea.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,167 @@
...
+* Fix an issue where if a Swift model class defined non-generic managed
+  properties after generic Realm properties (like `List<T>`), the schema
+  would be constructed incorrectly. Fixes an issue where creating such
+  models from an array could fail.
+* Loosen `RLMArray` and `RLMResults`'s generic constraint from `RLMObject` to
+  `NSObject`. This may result in having to add some casts to disambiguate
+  types.
+* Remove `RLMSyncPermissionResults`. `RLMSyncPermission`s are now vended out
+  using a `RLMResults`. This results collection supports all normal collection
+  operations except for setting values using KVO (since `RLMSyncPermission`s are
```
Comment (austinzheng) [inline CHANGELOG.md]: (totally my fault) I think this should be "KVC", not "KVO".

**Pull request merged**